### PR TITLE
[SV-COMP'18 17/19] Add option not to transform self-loops into assumes

### PR DESCRIPTION
**Pull request by tautschnig**
This allow disabling an optimisation in goto-symex
which is not compatible with termination checking.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]
LGTM modulo CI passing

File: src/jbmc/jbmc_parse_options.cpp [hunk 1]
```diff
@@ -154,6 +154,11 @@ void jbmc_parse_optionst::get_command_line_options(optionst &options)
   else
     options.set_option("propagation", true);
 
+  // transform self loops to assumptions
```
Comment (peterschrammel) [inline src/jbmc/jbmc_parse_options.cpp]: The documentation for this option in JBMC is not complete. Either remove (not really used anywhere at the moment) or make it complete.

**Review by peterschrammel — COMMENTED** [review]
Requires a rebase

File: src/jbmc/jbmc_parse_options.cpp (hunk 1 above)
Comment (tautschnig) [reply src/jbmc/jbmc_parse_options.cpp]: That's actually provided via `HELP_BMC` to all front-ends implementing BMC.

**Pull request merged**